operator/src/main/java/oracle/kubernetes/operator/helpers/KubernetesUtils.java

```diff
@@ -141,4 +142,26 @@ static V1ObjectMeta getResourceMetadata(Object resource) {
       return new V1ObjectMeta();
     }
   }
+
+  /**
+   * Returns true if the first metadata indicates a newer resource than does the second.
+   *
+   * @param first the first item to compare
+   * @param second the second item to compare
+   * @return true if the first object is newer than the second object
+   */
+  public static boolean isFirstNewer(V1ObjectMeta first, V1ObjectMeta second) {
+    if (second == null) return true;
+    if (first == null) return false;
+
+    DateTime time1 = first.getCreationTimestamp();
+    DateTime time2 = second.getCreationTimestamp();
+
+    if (time2.isAfter(time1)) {
+      return false;
+    }
+    return Integer.parseInt(second.getResourceVersion())
```

Comment:
i would prefer this to be written so the intent is clear and i dont have to rtfm to work out what it does

Reply:
Agreed, and will rewrite.